io.openems.edge.evcs.ocpp.server/src/io/openems/edge/evcs/ocpp/server/MyJsonServer.java

```diff
@@ -92,11 +107,22 @@ public void newSession(UUID sessionIndex, SessionInformation information) {
 
 				MyJsonServer.this.parent.ocppSessions.put(ocppIdentifier, sessionIndex);
 
-				var presentEvcss = MyJsonServer.this.parent.ocppEvcss.get(ocppIdentifier);
+				// Try to get the presentEvcss
+				List<AbstractManagedOcppEvcsComponent> presentEvcss = MyJsonServer.this.parent.ocppEvcss
+						.get(ocppIdentifier);
 
 				if (presentEvcss == null) {
+					// EVCS not yet configured, store session and retry later
```

Comment:
I'm concerned that there's a possibility of sending CALL messages simultaneously from the following three locations, making it difficult to meet the requirements of Section 4.1.1 (Synchronicity) in the specification. Therefore, I think it might be an option not to boot MyJsonServer until everything is properly set up. What do you think? @Sn0w3y 

- https://github.com/OpenEMS/openems/blob/3dc2d60b9c287dae767fdb9db8440b23848688e0/io.openems.edge.evcs.ocpp.server/src/io/openems/edge/evcs/ocpp/server/MyJsonServer.java#L104
- https://github.com/OpenEMS/openems/pull/2912/files#diff-a9d06149a614c84998d6bcad6d51f36ca40834f2e200120f2a2e9bfb290601eaR285
- https://github.com/OpenEMS/openems/blob/3dc2d60b9c287dae767fdb9db8440b23848688e0/io.openems.edge.evcs.ocpp.server/src/io/openems/edge/evcs/ocpp/server/EvcsOcppServer.java#L99

> 4.1.1. Synchronicity
>
> A Charge Point or Central System SHOULD NOT send a CALL message to the other party unless all the CALL messages it sent before have been responded to or have timed out. A CALL message has been responded to when a CALLERROR or CALLRESULT message has been received with the message ID of the CALL message.  
>  
> A CALL message has timed out when:  
>  
> • it has not been responded to, and  
> • an implementation-dependent timeout interval has elapsed since the message was sent.  
>  
> Implementations are free to choose this timeout interval. It is RECOMMENDED that they take into account the kind of network used to communicate with the other party. Mobile networks typically have much longer worst-case round-trip times than fixed lines.
>
> https://openchargealliance.org/download/7b06ab293c68fb6b4f4ae0960e502579c1c5516aa2b7acf0fdcedba585b9ea7f